### DISCUSSION TITLE
RX_standalone and TX_standalone Verses Python Scripts

### DISCUSSION THREAD

**Discussion by kenktk** [svpcom/wfb-ng#170] [Questions]
Hi,
I am new to wifibroadcast. Upon looking into the bash scripts for RX and the TX standalone, I saw that it was deprecated. I am confused on how I am supposed to use the Python scripts to send data. If I am supposed to run the python files to send and receive data, where do I configure the RX and TX ports (because there is a local.cfg and then there is a wifibroadcast.cfg in /etc/). 

This is all pretty confusing to me. My use case for wifibroadcast is to send telemetry only, (not video yet) between two android phones that are connected to two laptops. I have already established the tethering. Now all I have to do is set up bidirectional data streaming.

**Comment by svpcom** [comment]
Bash scripts are deprecated because it is hard to implement all WFB functions (TX diversity, mavlink aggregation, proxy and others) in bash.Python is not used on performance-critical path but for glue all together and for configuration management. 

How WFB configs works:
1. There are builtin config `telemetry/conf/master.cfg` than holds all defaults settings.
2. To override some of settings there are two ways:
   -  Create `/etc/wifibroadcast.cfg`
   -  Create `telemetry/conf/local.cfg`

First is used for production setups (when WFB is in /usr/bin) and second is for development and tests when you run WFB from sources

**Reply by svpcom** [reply]
For bidirectional telemetry it is hard to use standalone wfb_rx and wfb_tx because they unable to share the same UDP port (what python proxy does)

**Reply by kenktk** [reply]
Thank you for the quick reply! When I run any of the python scripts I get python errors. Would you suggest to rebuild?

**Reply by kenktk** [reply]
Errors such as:
`python3 version.py 
Traceback (most recent call last):
  File "version.py", line 28, in <module>
    main()
  File "version.py", line 13, in main
    ttuple = time.gmtime(int(sys.argv[1]))
IndexError: list index out of range
`
or when running:

`python3 server.py
Traceback (most recent call last):
  File "server.py", line 45, in <module>
    from .common import abort_on_crash, exit_status
ModuleNotFoundError: No module named '__main__.common'; '__main__' is not a package
`

Let me know if I need to file an issue. I am not sure exactly why this is happening.
